src/arrayMethodJoin.js

```diff
@@ -5,7 +5,22 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    const clearArr = this
+      .map((item) => item === null || item === undefined ? '' : item);
+    let joinSeparator = separator;
+    let joinString = clearArr[0];
+
+    if (this.length <= 1) {
+      return this.toString();
+    }
+
+    joinSeparator === undefined ? joinSeparator = ',' : joinSeparator += '';
```

Comment:
`joinSeparator = joinSeparator === undefined ?  ',' : joinSeparator + '';`
firstly you need to specify an element you're working with